tools/ncf_constraints.py

```diff
@@ -0,0 +1,107 @@
+import re
+import subprocess
+import json
+import os.path
+import shutil
+import sys
+import os
+import codecs
+
+if sys.version_info < (3,):
+  string_type = unicode
+else:
+  string_type = str
+
+### Constraint checking function
+
+def from_list( parameter, accepted_result):
+  return parameter in accepted_result
+
+def max_length( parameter, max_size):
+  return len(parameter) <= max_size
+
+def min_length( parameter, min_size):
+  return len(parameter) >= min_size
+
+def match_regex(parameter, regex):
+  match = re.search(regex, parameter)
+  return match is not None
+
+def not_match_regex(parameter, regex):
+  return not match_regex(parameter, regex)
+
+def allow_empty_string(parameter, allow):
+  return allow or not_match_regex(parameter, "^$")
+
+def allow_whitespace_string(parameter, allow):
+  return allow or not_match_regex(parameter, r'^\s') or not_match_regex(parameter, r'\s$')
```

Comment:
Exactly! Nice catch

Le 13 mai 2016 17:20:49 GMT+02:00, Jonathan Clarke <notifications@github.com> a écrit :
>> +
>> +def min_length( parameter, min_size):
>> +  return len(parameter) >= min_size
>> +
>> +def match_regex(parameter, regex):
>> +  match = re.search(regex, parameter)
>> +  return match is not None
>> +
>> +def not_match_regex(parameter, regex):
>> +  return not match_regex(parameter, regex)
>> +
>> +def allow_empty_string(parameter, allow):
>> +  return allow or not_match_regex(parameter, "^$")
>> +
>> +def allow_whitespace_string(parameter, allow):
>> +  return allow or not_match_regex(parameter, r'^\s') or
>not_match_regex(parameter, r'\s$')
>
>This does not do what you want it to: it will return "True" on the
>first value in the "or"-connected list that is True, therefore it
>succeeds (wrongly) on values like "bla " (because there is no space at
>the beginning)
>
>It should be:
>
>```
>return allow or (not_match_regex(parameter, r'^\s') and
>not_match_regex(parameter, r'\s$'))
>```
>
>---
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/Normation/ncf/pull/350/files/76b1c2546686678e1db499012c21a5911f5e1dc8#r63201268

-- 
Envoyé de mon téléphone Android avec K-9 Mail. Excusez la brièveté.